packages/frontend2/src/server/features/scaling/tvl/utils/get-scaling-summary-chart-data.ts

```diff
@@ -0,0 +1,86 @@
+import { type ValueRecord } from '@l2beat/database'
+import {
+  unstable_cache as cache,
+  unstable_noStore as noStore,
+} from 'next/cache'
+import { getEthPrices } from './get-eth-prices'
+import { type TvlProject, getTvlProjects } from './get-tvl-projects'
+import { getTvlValuesForProjects } from './get-tvl-values-for-projects'
+import { type TvlChartRange } from './range'
+import { sumValuesPerSource } from './sum-values-per-source'
+
+export async function getScalingSummaryChartData(
+  ...args: Parameters<typeof getCachedScalingChartData>
+) {
+  noStore()
+  return getCachedScalingChartData(...args)
+}
+
+type CommonOptions = {
+  range: TvlChartRange
+  excludeAssociatedTokens?: boolean
+}
+
+type DataType =
+  | {
+      type: 'layer2'
+    }
+  | {
+      type: 'projects'
+      projectIds: string[]
+    }
+
+export const getCachedScalingChartData = cache(
+  async ({
+    range,
+    excludeAssociatedTokens,
+    ...rest
+  }: CommonOptions & DataType) => {
+    const projectsFilter = (() => {
+      if (rest.type === 'layer2') {
+        return (project: TvlProject) =>
+          ['layer2', 'layer3'].includes(project.type)
```

Comment:
I dont like this approach. Let's say project.type is `"layer2" | "layer3"` and we refactor to `"Layer2" | "Layer3"`. In this case `if (project.type === 'layer2' || project.type === 'layer3')` would cause typescript to error but `.includes` would not. Another thing is that I don't know how JS handles temporary arrays and if it gets cleaned well. Overall I prefer more ifs.